sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestRecord.java

```diff
@@ -226,9 +228,19 @@ public long transportRequestId() {
     // region Methods
 
     public boolean expire() {
-        final GoneException error = new GoneException(this.toString(), null, this.args.physicalAddress());
-        BridgeInternal.setRequestHeaders(error, this.args.serviceRequest().getHeaders());
+        final CosmosException error;
+        if (this.args.serviceRequest().isReadOnly()) {
+            // Always converting RequestTimeoutException to GoneException for ReadOnly request. Since they can be safely retried.
+            error = new GoneException(this.toString(), null, this.args.physicalAddress());
+        } else if (!this.hasSendingRequestStarted()) {
+            // If request not sent for write request, converting to GoneException so it can be retried.
+            error = new GoneException(this.toString(), null, this.args.physicalAddress());
```

Comment:
On a side note can we merge these two if conditions? (exclusive of above comment which is a bigger change)

Reply:
yup, sure, will combine in the next iteration

Reply:
updated.